### PR TITLE
Fix pink GradientTexture2D

### LINKED ISSUE #93759: Creating a new Gradient2D resource shows up as pink
### Tested versions

4.3 beta1

### System information

Win 10

### Issue description

Creating a new GradientTexture2D creates a solid pink texture despite the gradient being the default black-white. Making any changes to the gradient fixes it.

image

### Steps to reproduce

New project > Add Sprite2D > Add new GradientTexture2D as the texture

### Minimal reproduction project (MRP)

N/A

### PR DESCRIPTION
**Pull request by matheusmdx**
Fix #93759, needs https://github.com/godotengine/godot/pull/94864 first.

This pr apply the suggested changes here: https://github.com/godotengine/godot/issues/93759#issuecomment-2207959523

### CODE REVIEW

**Comment by akien-mga** [comment]
There's the same `const_cast` for GradientTexture1D.

@KoBeWi added it in #81137, so it would be important to understand why.

**Comment by KoBeWi** [comment]
I think that `update_now()` is supposed to prepare the texture in case you want to get the image while an update is still queued.

**Comment by rune-scape** [comment]
i think that line of code probably does something important..
see my suggested fix: https://github.com/godotengine/godot/issues/93759#issuecomment-2207959523

**Comment by matheusmdx** [comment]
> see my suggested fix: [#93759 (comment)](https://github.com/godotengine/godot/issues/93759#issuecomment-2207959523)

I tried to apply this changes but i'm getting an error when i compile, most probably i'm doing something wrong:

<br>

gradient_texture.h:
Captura 2024-07-26 13-05-52-302452
Captura 2024-07-26 13-06-20-210124

<br>

gradient_texture.cpp:
Captura 2024-07-26 13-06-36-204074

Captura 2024-07-26 13-07-07-472456


Sorry if is some silly error, i'm don't have much knowledge of c++

**Comment by rune-scape** [comment]
ive made the same changes and didn't get any compiler error
the error output would be helpful,
keep in mind theres `GradientTexture1D` and `GradientTexture2D` and they both have a `const_cast`

**Comment by matheusmdx** [comment]
@rune-scape I pushed my changes for this pr can you check the tests results?

**Comment by rune-scape** [comment]
whoa, i swear i thought i compiled it, but ya im getting the same cryptic errors
i think theres a missing `create_custom_callable_function_pointer` function for const methods that return void .. in `callable_method_pointer.h`

**Comment by rune-scape** [comment]
made a PR (https://github.com/godotengine/godot/pull/94864) that fixes that problem and tested that these changes compile with that change

File: scene/resources/gradient_texture.cpp [hunk 1]
```diff
@@ -172,14 +172,13 @@ RID GradientTexture1D::get_rid() const {
 }
 
 Ref<Image> GradientTexture1D::get_image() const {
-	const_cast<GradientTexture1D *>(this)->update_now();
```
Comment (rune-scape) [inline scene/resources/gradient_texture.cpp:175]: this and the other call to `update_now()` should still be here, just as a regular method call without the `const_cast`

**Comment by matheusmdx** [comment]
Changes done, i also updated the pr description to inform about https://github.com/godotengine/godot/pull/94864

**Comment by akien-mga** [comment]
Can be rebased as #94864 was just merged.

**Comment by matheusmdx** [comment]
Done

**Review by akien-mga — APPROVED** [review]

**Comment by akien-mga** [comment]
Thanks!

**Pull request merged**